src/pydicom/pixels/decoders/base.py

```diff
@@ -1198,11 +1226,19 @@ def _as_array_native(runner: DecodeRunner, index: int | None) -> "np.ndarray":
                     "original buffer for bit-packed pixel data"
                 )
 
-            length_pixels = runner.frame_length(unit="pixels")
             if index is None:
                 length_pixels *= runner.number_of_frames
 
-            return np.unpackbits(arr, bitorder="little", count=length_pixels)
+            unpacked = np.unpackbits(
+                arr, bitorder="little", count=ceil(length_bytes) * 8
+            )
+
+            # May need to remove bits from the beginning or end if frame
+            # boundaries are not byte-aligned
+            if bit_offset_start is not None or bit_offset_end is not None:
+                unpacked = unpacked[bit_offset_start:bit_offset_end]
```

Comment:
Can you not just replace `[bit_offset_start : bit_offset_end]` with `[bit_offset_start : bit_offset_start + length_pixels]`?

Reply:
Addressed in [896c6b1](https://github.com/pydicom/pydicom/pull/2140/commits/896c6b1fc48f431d63e60d944aedd6d53a947293)